### ISSUE TITLE
Issue calling macro functions from within macros

### ISSUE THREAD

**Issue by nibblebits** [nibblebits/DragonCompiler#42]
```
#define ABC_SUM(x, y) x + y

#define ABC ABC_SUM(5, 5)
#if ABC

#endif

```

**Comment by nibblebits** [comment]
rogram received signal SIGSEGV, Segmentation fault.
0x00005555555605d8 in is_operator_token (token=0x0) at helper.c:300
300	    return token->type == TOKEN_TYPE_OPERATOR;
(gdb) bt
#0  0x00005555555605d8 in is_operator_token (token=0x0) at helper.c:300
#1  0x00005555555588ce in expressionable_deal_with_additional_expression (
    expressionable=0x5555555af5f0) at expressionable.c:183
#2  0x0000555555558a06 in expressionable_parse_parentheses (
    expressionable=0x5555555af5f0) at expressionable.c:218
#3  0x0000555555558c93 in expressionable_parse_exp (

**Comment by nibblebits** [comment]
Fixed: cf8ed4640bf7b2a187b83fbb6727038961b2101b
